sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -165,6 +166,7 @@ case class FileSourceScanExec(
     requiredSchema: StructType,
     partitionFilters: Seq[Expression],
     optionalBucketSet: Option[BitSet],
+    optionalNumCoalescedBuckets: Option[Int],
```

Comment:
I think we can have coalesced bucket info in explain. Looks like we still show original bucket numbers.

Reply:
Yes, good idea. @cloud-fan/@maropu Do you want me to include this in this PR or do it as a follow up since this PR is already approved? I am fine with either one. Thanks @viirya!

Reply:
It's easy to fix: just update `FileSourceScanExec.metadata`. Let's do it in this PR

Reply:
Will do.